### ISSUE TITLE
Safari 14.0.3: cannot start playback of recorded video

### ISSUE THREAD

**Issue by thijstriemstra** [collab-project/videojs-record#552]
> When trying to render the recorded video, the live stream is rendered, not the recorded one.

Tested in following browsers:

| OS | Browser | Working |
| --- | --- | --- |
| MacOS 11.2.2 | Safari 14.0.3 | No |
| MacOS 11.3 | Safari 14.1 | Yes |
| MacOS 11.2.2 | Safari Tech Preview 121 | Yes |
| iOS 14.4 | Mobile Safari 14.0.3 | No |
| iOS 14.5 | Mobile Safari 14.1 | Almost |

`ffprobe` reports:
```
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from '/Users/foo/Downloads/test.mp4':
  Metadata:
    major_brand     : iso5
    minor_version   : 1
    compatible_brands: isomiso5hlsf
    creation_time   : 2021-03-06T21:08:37.000000Z
  Duration: 00:00:03.21, start: 0.000000, bitrate: 2533 kb/s
  Stream #0:0(und): Video: h264 (High) (avc1 / 0x31637661), yuv420p(progressive), 640x480 [SAR 1:1 DAR 4:3], 2501 kb/s, 23.61 fps, 600 tbr, 600 tbn, 1200 tbc (default)
    Metadata:
      creation_time   : 2021-03-06T21:08:37.000000Z
      handler_name    : Core Media Video
      vendor_id       : [0][0][0][0]
  Stream #0:1(und): Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, mono, fltp, 176 kb/s (default)
    Metadata:
      creation_time   : 2021-03-06T21:08:37.000000Z
      handler_name    : Core Media Audio
      vendor_id       : [0][0][0][0]
```

**Comment by acraig39** [comment]
I have everything working great in Chrome but in Safari (14.0.3) in addition to above issue I'm also finding -
1) that the camera doesn't turn off after recording; 
2) the file's mime type shows as application/octet-stream rather than video/mp4 say; 
3) the size of video file is much larger than when I use chrome and get a webm file; 
4) it is unable to record as webm with an error in the console saying - MediaRecorder API seems unable to record mimeType: – "video/webm"
Not sure if these should be raised separately but I'm new to this so apologies. I'm just very keen to get the same functionality working in Safari as what I've had in Chrome for nearly a year now!
btw in Safari Preview 121 it does show the recorded video ok and not the live stream.

**Comment by thijstriemstra** [comment]
I searched for Safari workarounds in video.js and found this in the `Player.play()` [implementation](https://github.com/videojs/video.js/blob/b13cb07d5b6d1f3f28ed355e12da40d178131df8/src/js/player.js#L2347): 

```javascript
// if we are in Safari, there is a high chance that loadstart will trigger
// after the gesture timeperiod in that case, we need to prime the video
// element by calling load so it'll be ready in time
if (!isSrcReady && (browser.IS_ANY_SAFARI || browser.IS_IOS)) {
    this.load();
}
```

Likely it's the thing breaking Safari 14.0.3..

**Comment by thijstriemstra** [comment]
> that the camera doesn't turn off after recording;

It doesn't do this by default.

> the file's mime type shows as application/octet-stream rather than video/mp4 say;

No, it shows as video/mp4 here.

> the size of video file is much larger than when I use chrome and get a webm file;

Yea, Safari doesn't use VP8 or VP9 codec (but h264) resulting in a bigger file (and better quality in theory).

> MediaRecorder API seems unable to record mimeType: – "video/webm"

That's a warning, meaning it will use `video/mp4` instead.

> I'm new to this so apologies. 

Yea, better to open a new ticket if issue's still there after this ticket is marked as resolved.

**Comment by thijstriemstra** [comment]
Looks like this is a bug in Webkit/Safari 14.0.3 (unfortunately). Good news is that someone at Apple tested with videojs-record (very cool btw), [reported the Webkit issue](https://bugs.webkit.org/show_bug.cgi?id=219258), fixed it and it's released in Safari Technology Preview 121, but this is not yet released in a new Safari version.

See webkit commit referencing videojs-record: https://trac.webkit.org/changeset/270171/webkit

Commit msg:

>  ​https://collab-project.github.io/videojs-record/demo/video-only.html is not working
> ​https://bugs.webkit.org/show_bug.cgi?id=219258
> <rdar://problem/69759808>
>
> Reviewed by Eric Carlson.
>
> Source/WebCore:
>
> Make 'HTMLMediaElement.srcObject = blob' functional by making sure to register a blob URL before trying to load.
> Clean-up internal slots when srcObject is set to make sure we do not load a MediaStream even if the current MediaProvider is a Blob.
>
> Fix a Use-After-Move bug that can be triggered in case of synchronous media loading failure.

**Comment by thijstriemstra** [comment]
> Looks like this is a bug in Webkit/Safari 14.0.3 unfortunately.

Tested with new Desktop Safari 14.1 and seems this is fixed 🥳  

Also tested it with Mobile Safari 14.1 on iOS 14.5 and it worked: I was able to start playback of the recorded video but the first frame of the video was not visible after pressing stop recording. I will open a new, separate ticket for that.

I updated [the table](https://github.com/collab-project/videojs-record/issues/552#issue-821123730) listed in this ticket.
